src/coreclr/vm/ilmarshalers.cpp

```diff
@@ -4296,31 +4294,17 @@ void ILNativeArrayMarshaler::EmitNewSavedSizeArgLocal(ILCodeStream* pslILEmit)
     pslILEmit->EmitSTLOC(m_dwSavedSizeArg);
 }
 
-
-FCIMPL4(void, MngdNativeArrayMarshaler::CreateMarshaler, MngdNativeArrayMarshaler* pThis, MethodTable* pMT, UINT32 dwFlags, PCODE pManagedMarshaler)
+extern "C" void QCALLTYPE MngdNativeArrayMarshaler_ConvertSpaceToNative(MngdNativeArrayMarshaler* pThis, QCall::ObjectHandleOnStack pManagedHome, void** pNativeHome)
 {
-    FCALL_CONTRACT;
+    QCALL_CONTRACT;
 
-    // Don't check whether the input values are negative - passing negative size-controlling
-    // arguments and compensating them with a positive SizeConst has always worked.
-    pThis->m_pElementMT            = pMT;
-    pThis->m_vt                    = (VARTYPE)(dwFlags);
-    pThis->m_NativeDataValid       = (BYTE)((dwFlags & FLAG_NATIVE_DATA_VALID) != 0);
-    dwFlags &= ~FLAG_NATIVE_DATA_VALID;
-    pThis->m_BestFitMap            = (BYTE)(dwFlags >> 16);
-    pThis->m_ThrowOnUnmappableChar = (BYTE)(dwFlags >> 24);
-    pThis->m_Array                 = TypeHandle();
-    pThis->m_pManagedMarshaler     = pManagedMarshaler;
-}
-FCIMPLEND
-
-FCIMPL3(void, MngdNativeArrayMarshaler::ConvertSpaceToNative, MngdNativeArrayMarshaler* pThis, OBJECTREF* pManagedHome, void** pNativeHome)
-{
-    FCALL_CONTRACT;
+    BEGIN_QCALL;
 
-    HELPER_METHOD_FRAME_BEGIN_0();
+    GCX_COOP();
 
-    BASEARRAYREF arrayRef = (BASEARRAYREF) *pManagedHome;
+    BASEARRAYREF arrayRef = NULL;
+    GCPROTECT_BEGIN(arrayRef);
```

Comment:
Nit: You do not need to GCPROTECT the copy of arg in situations like this. It is cheaper to do `pManagedHome.Get` twice instead.